idea/src/org/jetbrains/kotlin/idea/inspections/ConvertToObjectDeclarationInspection.kt

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2010-2019 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license
+ * that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.idea.inspections
+
+import com.intellij.codeInspection.LocalQuickFix
+import com.intellij.codeInspection.ProblemDescriptor
+import com.intellij.codeInspection.ProblemsHolder
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiElementVisitor
+import com.intellij.psi.search.searches.ReferencesSearch
+import org.jetbrains.kotlin.descriptors.ClassConstructorDescriptor
+import org.jetbrains.kotlin.idea.caches.resolve.analyze
+import org.jetbrains.kotlin.js.resolve.diagnostics.findPsi
+import org.jetbrains.kotlin.lexer.KtTokens
+import org.jetbrains.kotlin.psi.*
+import org.jetbrains.kotlin.psi.synthetics.findClassDescriptor
+import org.jetbrains.kotlin.resolve.descriptorUtil.getSuperClassNotAny
+import org.jetbrains.kotlin.resolve.descriptorUtil.getSuperInterfaces
+
+class ConvertToObjectDeclarationInspection : AbstractKotlinInspection() {
+    override fun buildVisitor(holder: ProblemsHolder, isOnTheFly: Boolean): PsiElementVisitor {
+        return classVisitor { clazz ->
+            val descriptor = clazz.findClassDescriptor(clazz.analyze())
+            if (descriptor.getSuperClassNotAny() != null || descriptor.getSuperInterfaces().isNotEmpty()) return@classVisitor
+            if (descriptor.constructors.any { it.hasConstructorCallUsage() }) return@classVisitor
+            val declarations = clazz.declarations
+            if (declarations.size == 1) {
+                val firstDecl = declarations[0] as? KtObjectDeclaration ?: return@classVisitor
+                if (firstDecl.isCompanion()) {
+                    holder.registerProblem(
```

Comment:
Should we highlight the whole class? It will be enough to highlight only `class Name`.